client/client_test.go

```diff
@@ -246,14 +246,16 @@ func testIntegration(t *testing.T, funcs ...func(t *testing.T, sb integration.Sa
 		}),
 	)
 
-	integration.Run(t, integration.TestFuncs(
-		testBridgeNetworkingDNSNoRootless,
-	),
-		mirrors,
-		integration.WithMatrix("netmode", map[string]interface{}{
-			"dns": bridgeDNSNetwork,
-		}),
-	)
+	if runtime.GOOS != "windows" {
```

Comment:
Each individual test in `testBridgeNetworkingDNSNoRootless` should instead declare that it cannot be run in windows, right?

I want to avoid as many explicit `runtime.GOOS != "windows"` checks as possible, they're a little inflexible, and complicate reading the code.

Reply:
For some reason, even with the line to skip at the beginning of the function, some set up was still being done; so I'd to guard this. Any tips will be appreciated.

Here's the trace:
```
        	Error Trace:	C:/buildkit/util/testutil/integration/run.go:198
        	Error:      	Received unexpected error:
        	            	process exited: C:\Users\Administrator\go\bin\buildkitd.exe --containerd-worker-gc=false --containerd-worker=true --containerd-worker-addr npipe:////./pipe/containerd-bktest_containerd2165329572 --containerd-worker-labels=org.mobyproject.buildkit.worker.sandbox=true --config=C:\Users\ADMINI~1\AppData\Local\Temp\2\bktest_config1965241481\buildkitd.toml --root C:\Users\ADMINI~1\AppData\Local\Temp\2\bktest_buildkitd2000050298 --addr npipe:////./pipe/buildkitd-bktest_buildkitd2000050298 --debug
        	            	github.com/moby/buildkit/util/testutil/integration.WaitSocket
        	            		C:/buildkit/util/testutil/integration/util.go:111
        	            	github.com/moby/buildkit/util/testutil/workers.runBuildkitd
        	            		C:/buildkit/util/testutil/workers/util.go:93
        	            	github.com/moby/buildkit/util/testutil/workers.(*Containerd).New
        	            		C:/buildkit/util/testutil/workers/containerd.go:228
        	            	github.com/moby/buildkit/util/testutil/integration.newSandbox
        	            		C:/buildkit/util/testutil/integration/sandbox.go:100
        	            	github.com/moby/buildkit/util/testutil/integration.Run.func3.1
        	            		C:/buildkit/util/testutil/integration/run.go:197
        	            	testing.tRunner
        	            		C:/Program Files/Go/src/testing/testing.go:1595
        	            	runtime.goexit
        	            		C:/Program Files/Go/src/runtime/asm_amd64.s:1650
        	            	creating worker
        	            	github.com/moby/buildkit/util/testutil/integration.newSandbox
        	            		C:/buildkit/util/testutil/integration/sandbox.go:102
        	            	github.com/moby/buildkit/util/testutil/integration.Run.func3.1
        	            		C:/buildkit/util/testutil/integration/run.go:197
        	            	testing.tRunner
        	            		C:/Program Files/Go/src/testing/testing.go:1595
        	            	runtime.goexit
        	            		C:/Program Files/Go/src/runtime/asm_amd64.s:1650
        	Test:       	TestIntegration/TestBridgeNetworkingDNSNoRootless/worker=containerd/netmode=dns
    --- FAIL: TestIntegration/TestBridgeNetworkingDNSNoRootless/worker=containerd/netmode=dns (20.22s)
    ```